eos_data_distribution/ndn/chunks.py

```diff
@@ -125,12 +125,15 @@ def _schedule_interests(self):
 
             self._request_segment(next_segment)
 
+    def _request(self, name):
+        self.expressInterest(name, forever=True)
+
     def _request_segment(self, n):
         ndn_name = Name(self.name).appendSegment(n)
-        self.expressInterest(ndn_name, forever=True)
         if self._segments is not None:
             self._segments[n] = SegmentState.OUTGOING
         self._num_outstanding_interests += 1
+        self._request(ndn_name)
```

Comment:
`self._request` already is a fairly simple method that I don't see much value in extracting. Requesting a barename and requesting a chunk are two very different things IMO.